api/python/tests/integration/test_packages.py

```diff
@@ -1410,3 +1416,70 @@ def test_resolve_hash(self):
 
         with pytest.raises(QuiltException, match='Found multiple matches'):
             Package.resolve_hash(pkg_name, LOCAL_REGISTRY, hash_prefix)
+
+
+# The following tests were moved out of the PackageTest class to enable parametrization.
+# see (https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses)
+@pytest.mark.parametrize(
+    'update_policy, expected_size, expected_meta, expected_hash',
+    [('incoming', 6, {'name': 'incoming'}, None), ('existing', 1, {'name': 'existing'}, None)]
+)
+def test_set_dir_update_policy(update_policy, expected_size, expected_meta, expected_hash):
+    """Verify building a package with update policy. """
+
+    data_dir = pathlib.Path(__file__).parent / "data"
+    nested_dir = data_dir / 'nested'
+    nested_dir_2 = data_dir / 'nested2'
+    pkg = Package()
+    pkg = pkg.set_dir("/", nested_dir, meta={'name': 'test_meta'})
+    assert 'one.txt' in pkg.keys()
+    assert pkg.meta == {'name': 'test_meta'}
+
+    pkg = pkg.set_dir("/", nested_dir_2, meta={'name': update_policy}, update_policy=update_policy)
+    assert 'one.txt' in pkg.keys()
+    assert pkg['one.txt'].size == expected_size
+    assert pkg['one.txt'].hash == expected_hash
+    assert pkg.meta == expected_meta
```

Comment:
you're doing the right stuff here but our test folders should really have multiple files so that we are testing and actual merge. and your tests should check that some keys changed and others did not.